### DISCUSSION TITLE
Manifesto for a from-scratch bike touring hospex application

### DISCUSSION THREAD

**Discussion by lennartvanlaake** [OpenHospitalityNetwork/fedi-trustroots#85] [General]
In last Monday's discussion I argued for creating a new platform for the cycle touring community rather than adjusting the Trustroots codebase. I would like to use this threat to further discuss this concept and to validate whether what I have in mind makes sense.

### Complement, not copy

If we are making a new platform we should focus on features that other platforms don't have or don't do well.

### A-symmetric "federation"

Use "federation"/inter-platform-communication features to make it attractive for other (bike touring) platforms to integrate with the new one and to compensate for the features that the new platform does not do well.

### KISS (Keep It Simple Stupid)

Don't add any features that are not strictly necessary. Everything that we do should be done in the simpelest way possible that still provides value to end-users.

### Least amount of clicks to get a bed

We should focus on the perspective of a tired cyclist who just wants to sleep/meet some people. That cyclists (probably) wants the UI to make it extremely easy and clear which buttons to press to get a bed.

### "Indirect"/One-to-many requests only

My idea would be to only have indirect hosting requests. A cylist would say: I will be in Vienna tomorrow. Hosts/other cyclists within a certain range will get a notification of this.

Instead of spamming everyone in Vienna separately, the application could keep track of the indirect host request and the requester could disable the request once a host has been found.

### Cyclists should be able to see other cylists

If you publish that you want to meet people in Vienna tomorrow, other cyclists should be able to see you on the map or get notified if they are in the neighbourhood. 

### No messaging, no (elaborate) profiles

Messaging and profiles tend to be a large part of the codebase of hospex platforms. Not having these features would save a lot of code. 

To replace messaging we can allow users to grant access contact details with a button-press (user A has requested to get your contact details. Will you allow user A to see them?). 

People have lots of profiles on the internet (social media, other hospex platforms, their own websites). So if you really want to see someone's profile picture, just link to their profile somewhere else. Ideally we could integrate this with other hospex platforms so you can see (for example) someone's Trustroots profile without logging into Trustroots.

### App-first

A big problem for cyclists is that the responses tend to be pretty slow. Most people use their mobile phone for quick notifications. If we build the app first (or only an app), we would be able to push native notifications and have a feature that other platforms do not have. This might make this platform more appealing to federate with.

## Pitch

I think this platform would be more attractive because it provides something new to end users immediately, without them having to care about federation. By excluding elaborate profiles and messaging from the scope, I believe we would be able to build a meaningful MVP in a relatively short period of time (I'd say a couple of months).  

To get other communities/platforms excited about federation I think this approach would help. If this platform just wants access to the locations of their hosts, that should be way simpeler for them to implement then a fully symmetrical federation. Since lots of platforms sturggle with mobile notifications, being able to provide that functionality to other platforms seems like a large selling point of federating.

Let me know if you have any questions/comments/additions on this plan :)

**Comment by mariha** [comment]
Cool, for the reference (and inclusion in the discussion), I am linking:
- Pitch for [Open and free bicycle hospitality exchange service](https://warmshowers.bike/survey) that @saemy wrote and our [message to the community](https://warmshowers.bike/)
- Pitch for federation that @mariha (me!) happened to write (not sure if it’s worth reading) - OpenHospitalityNetwork/OpenHospitalityNetwork.github.io#9
- [Pondering about federation in SocialHub](https://socialhub.activitypub.rocks/t/hospitality-exchange-community-considers-moving-to-the-fediverse/1765/8) that @nicksellen wrote
